Rust 1.41 compiler error with rust-numpy 0.14.0

While trying to upgrade retworkx to pyo3 0.14 and numpy 0.14.0 I'm hitting a compiler error on the msrv job that builds with rust 1.41. https://github.com/Qiskit/retworkx/pull/384/checks?check_run_id=3047301998#step:7:275

```
error[E0658]: use of unstable library feature 'ptr_offset_from'
    --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/numpy-0.14.0/src/convert.rs:70:31
     |
  70 |             unsafe { orig_ptr.offset_from(vec.as_ptr()) as usize }
     |                               ^^^^^^^^^^^
     |
     = note: for more information, see https://github.com/rust-lang/rust/issues/41079
  
  error: aborting due to previous error
  
  For more information about this error, try `rustc --explain E0658`.
  error: could not compile `numpy`.
```

It looks like that api wasn't stabilized until rust 1.47.0. I'm not sure why the msrv CI job here didn't catch this though.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -83,6 +83,7 @@ jobs:
         with:
           profile: minimal
           toolchain: 1.41.1
+          default: true
       - name: Install maturin, poetry, and toml
         run: pip install maturin poetry toml
       - name: Create an isolated example directory

--- a/src/convert.rs
+++ b/src/convert.rs
@@ -62,12 +62,12 @@ where
         let (strides, dim) = (self.npy_strides(), self.raw_dim());
         let orig_ptr = self.as_ptr();
         // Element of which size is 0 is not supported, but check it for future changes
-        let is_empty_or_size0 = self.is_empty() || std::mem::size_of::<Self::Item>() == 0;
+        let is_empty_or_size0 = self.is_empty() || std::mem::size_of::<A>() == 0;
         let vec = self.into_raw_vec();
         let offset = if is_empty_or_size0 {
             0
         } else {
-            unsafe { orig_ptr.offset_from(vec.as_ptr()) as usize }
+            (orig_ptr as usize - vec.as_ptr() as usize) / std::mem::size_of::<A>()
         };
         let mut boxed_slice = vec.into_boxed_slice();
         // data_ptr is not always the pointer to the 1st element.

